### PR TITLE
GPII-1810: Add ability to configure `jsonify` helper...

### PR DESCRIPTION
**Pull request by the-t-in-rtf**
See [GPII-1810](https://issues.gpii.net/browse/GPII-1810) for details.

### CODE REVIEW

**Comment by amb26** [comment]
The browser-renderer-standalone-tests.js test hangs for me with the following:

21:26:33.262:  Test case listener has not responded after 5000ms - at sequence pos 4 of 7 sequence element {
    "event": "{testEnvironment}.browser.events.onLoaded",
    "listener": "{testEnvironment}.browser.evaluate",
    "args": [
        { Function

```
    },
    "md",
    "[unified listing](http://ul.gpii.net/)"
]
```

} of fixture Confirm that the client-side renderer can render markdown...

Environment;: Windows 7 64bit with node.js 4.4.7 32-bit

File: .qi.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+app_name: gpii-handlebars
```
Comment (amb26) [inline .qi.yml]: Can you comment these new mysterious files in order to explain their purpose and provenance?

File: Gruntfile.js [hunk 2]
```diff
@@ -1,22 +1,18 @@
+/* eslint-env node */
```
Comment (amb26) [inline Gruntfile.js:1]: See notes for gpii-binder Gruntfile

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,131 @@
+# -*- mode: ruby -*-
```
Comment (amb26) [inline Vagrantfile:1]: Add some hints as to where this file comes from, what it does, and how we plan to eliminate in time its cut and pasting to all our projects

File: docs/handlebars.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+1. `layouts`
+2. `pages`
+3. `partials`
+
+The handlebars middleware supports inheritance between multiple view directories.  Layouts, pages and partials will be
+rendered from the first matching view directory.
+
+# Adding block helpers
+
+Child components of this grade that extend the [`gpii.express.helper`](helper.md) grade are made available as block
```
Comment (amb26) [inline docs/handlebars.md:50]: Add link to explain what block helpers are here

File: docs/helper.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+### `{that}.getHelper()`
+
+Each "helper" module is expected to implement a `getHelper` invoker with an invoker that returns a helper function. something like the following:
+
+```
+fluid.registerNamespace("your.helper");
+
+your.helper.getHelperFn = function(that){
+   return function(arg1, arg2) {
```
Comment (amb26) [inline docs/helper.md:16]: This is rather peculiar as an example - does this represent something that every helper would need to deal with, or is it really just an arbitrary example? Is there any significance to "fn"?

**Comment by the-t-in-rtf** [comment]
In reply to [Antranig's comment about the failing renderer tests](https://github.com/GPII/gpii-handlebars/pull/20#issuecomment-234738343).  I will investigate this week while updating to use `gpii-webdriver` instead of `gpii-test-browser`.

File: Gruntfile.js (hunk 2 above)
Comment (the-t-in-rtf) [reply Gruntfile.js:1]: This has been addressed throughout "the stack".

File: Vagrantfile (hunk 3 above)
Comment (the-t-in-rtf) [reply Vagrantfile:1]: I have added comments about the QI files throughout "the stack".  See [GPII-1988](https://issues.gpii.net/browse/GPII-1988).

File: .qi.yml (hunk 1 above)
Comment (the-t-in-rtf) [reply .qi.yml]: Done, throughout "the stack".

**Comment by the-t-in-rtf** [comment]
Now that this has been updated to use a later version of `gpii-webdriver`, the tests pass again.  This should be ready for review.

**Comment by the-t-in-rtf** [comment]
FYI, we cannot upgrade beyond version 1.1.2 of express-handlebars until this issue is resolved:

https://github.com/ericf/express-handlebars/issues/138

That's the only dependency that's not up to date.

**Comment by the-t-in-rtf** [comment]
Hold off on the review, I just found a TODO with a bunch of browser tests that still need to be converted to gpii-webdriver.  I will finish those hopefully Monday.

**Comment by the-t-in-rtf** [comment]
All browser tests have been converted, this is ready for review.

**Comment by the-t-in-rtf** [comment]
Although it is certainly safe to review, this should not be merged/published until the following dependencies have been merged/published:
- [gpii-binder](https://github.com/GPII/gpii-binder/pull/3)
- [gpii-express](https://github.com/GPII/gpii-express/pull/25)
- [gpii-webdriver](https://github.com/GPII/gpii-webdriver/pull/1)

**Comment by amb26** [comment]
I get a test failure 01:05:35.987:  jq: FAIL: Module "chrome: Testing `initBlock` helper..." Test name "chrome: Confirm the page was rendered
 and that the initBlock component was created correctly..." - Message: Died on test #1     at Object.QUnit.asyncTest (E:
\Source\gits\gpii-handlebars\node_modules\infusion\tests\lib\qunit\js\qunit.js:401:9)
    at E:\Source\gits\gpii-handlebars\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:621:29
    at Object.fluid.each (E:\Source\gits\gpii-handlebars\node_modules\infusion\src\framework\core\js\Fluid.js:502:17)
    at Object.fluid.test.processTestCase (E:\Source\gits\gpii-handlebars\node_modules\infusion\tests\test-core\utils\js\
IoCTestUtils.js:594:15)
    at E:\Source\gits\gpii-handlebars\node_modules\infusion\tests\test-core\utils\js\IoCTestUtils.js:644:24
    at Object.fluid.each (E:\Source\gits\gpii-handlebars\node_modules\infusion\src\framework\core\js\Fluid.js:502:17)
    at Object.fluid.test.processTestCaseHolder (E:\Source\gits\gpii-handlebars\node_modules\infusion\t .... [output supp
ressed at 1024 chars - for more output, increase fluid.logObjectRenderChars]
01:05:36.045:  jq: Source: Error: The ChromeDriver could not be found on the current PATH. Please download the latest ve
rsion of the ChromeDriver from http://chromedriver.storage.googleapis.com/index.html and ensure it can be found on your
PATH.

Is this 3rd-party library required for the tests? It's odd that it appears on just this test. If required, should be added to the README

**Comment by the-t-in-rtf** [comment]
Previously, this was a bit hidden, as firefox "just works", and was the option used by default on Windows. I have a dev dependency that will install the required driver, I'd like to automate that install and setting the correct path.

File: docs/helper.md (hunk 5 above)
Comment (the-t-in-rtf) [reply docs/helper.md:16]: The handlebars API expects to be given a function with one of a few acceptable signatures.  I could see moving to the same type of approach we used with express middleware, i.e. requiring a simpler invoker to be defined and handling the wrapping separately.

Comment (the-t-in-rtf) [reply docs/helper.md:16]: Given that I am the only one who has written these to date, it is fairly low risk.  I believe I'd just need to update this package and the "deps" helper in the sandbox.

File: .gitignore [hunk 6]
```diff
@@ -4,3 +4,4 @@ bower_components
 coverage
 reports
 .vagrant
+yarn.lock
```
Comment (colinbdclark) [inline .gitignore]: Are you sure you want to omit your yarn lock file? https://yarnpkg.com/en/docs/yarn-lock#toc-check-into-source-control

Comment (the-t-in-rtf) [reply .gitignore]: Ah, thanks.  Hadn't read that. I will scrub my gitignores globally and commit.

**Comment by the-t-in-rtf** [comment]
@amb26, once the gpii-pouchdb and/or gpii-json-schema pulls are merged, this is a good candidate to review next.  I have already updated the deps (again) and confirmed the tests working on Windows.

**Pull request merged**